File: src/main/old/bbarena/model/event/roster/AddPlayerExperienceEvent.java
package bbarena.model.event.roster;

import bbarena.model.Arena;
import bbarena.model.team.Experience;
import bbarena.model.team.Player;
import bbarena.model.util.Concat;
import bbarena.model.util.Pair;

public class AddPlayerExperienceEvent extends RosterEvent {

	private static final long serialVersionUID = 6408478747912466903L;

	private Player _player = null;
	private Experience _newExperience = null;

	public AddPlayerExperienceEvent(long playerId, int comp, int td, int inter, int cas, int mvp, int old, int scars, int fouls, int rush) {
		super(playerId);
		_newExperience = new Experience(comp, td, inter, cas, mvp, old, scars, fouls, rush);
	}

	@Override
	public void doEvent(Arena arena) {
		_player = getPlayer(arena);
		arena.getPlayerManager().addPlayerExperience(_player, _newExperience);
	}

	@Override
	public void undoEvent(Arena arena) {
		arena.getPlayerManager().subPlayerExperience(_player, _newExperience);
	}

	@Override
	public String getString() {
		return Concat.INSTANCE.buildLog(getClass(),
				new Pair("playerId", getPlayerId()),
				new Pair("exp", _newExperience));
	}

	public Experience getNewExperience() {
		return _newExperience;
	}

}

File: src/main/kotlin/bbarena/model/event/roster/AddPlayerExperienceEvent.kt
package bbarena.model.event.roster

import bbarena.model.Arena
import bbarena.model.team.Experience
import bbarena.model.team.Player
import bbarena.model.util.Concat

class AddPlayerExperienceEvent(playerId: Long, val comp: Int, val td: Int, val inter: Int, val cas: Int, val mvp: Int, val old: Int, val scars: Int, val fouls: Int, val rush: Int) : RosterEvent(playerId) {

    private var player: Player? = null
    var newExperience: Experience = Experience(comp, td, inter, cas, mvp, old, scars, fouls, rush)

    override fun doEvent(arena: Arena) {
        player = getPlayer(arena)
        val p = player
        if (p != null) {
            arena.playerManager.addPlayerExperience(p, newExperience)
        }
    }

    override fun undoEvent(arena: Arena) {
        val p = player
        if (p != null) {
            arena.playerManager.subPlayerExperience(p, newExperience)
        }
    }

    override val string: String
        get() = Concat.buildLog(javaClass,
                Pair("playerId", playerId),
                Pair("exp", newExperience))

}
